aea/cli/upgrade.py

```diff
@@ -0,0 +1,203 @@
+"""Implementation of the 'aea upgrade' subcommand."""
+
+from typing import cast
+
+import click
+
+from aea.cli.add import add_item
+from aea.cli.registry.add import _get_package_meta
+from aea.cli.remove import remove_item
+from aea.cli.utils.click_utils import PublicIdParameter
+from aea.cli.utils.context import Context
+from aea.cli.utils.decorators import check_aea_project, clean_after, pass_ctx
+from aea.cli.utils.package_utils import (
+    find_item_locally,
+    get_item_public_id_by_author_name,
+    get_items,
+    is_item_present,
+)
+from aea.configurations.base import PublicId
+
+
+@click.group(invoke_without_command=True)
+@click.option("--local", is_flag=True, help="For upgrading from local folder.")
+@click.pass_context
+@check_aea_project
+def upgrade(click_context, local):
+    """Upgrade agent's component."""
+    ctx = cast(Context, click_context.obj)
+    if local:
+        ctx.set_config("is_local", True)
+
+    if click_context.invoked_subcommand is None:
+        upgrade_project(ctx)
+
+
+@upgrade.command()
+@click.argument("connection_public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def connection(ctx: Context, connection_public_id: PublicId):
+    """Upgrade a connection at the configuration file."""
+    upgrade_item(ctx, "connection", connection_public_id)
+
+
+@upgrade.command()
+@click.argument("contract_public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def contract(ctx: Context, contract_public_id: PublicId):
+    """Upgrade a contract at the configuration file."""
+    upgrade_item(ctx, "contract", contract_public_id)
+
+
+@upgrade.command()
+@click.argument("protocol_public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def protocol(ctx: Context, protocol_public_id):
+    """Upgrade a protocol for the agent."""
+    upgrade_item(ctx, "protocol", protocol_public_id)
+
+
+@upgrade.command()
+@click.argument("skill_public_id", type=PublicIdParameter(), required=True)
+@pass_ctx
+def skill(ctx: Context, skill_public_id: PublicId):
+    """Upgrade a skill for the agent."""
+    upgrade_item(ctx, "skill", skill_public_id)
+
+
+@clean_after
+def upgrade_project(ctx: Context) -> None:  # pylint: disable=unused-argument
+    """Perform project upgrade."""
+    click.echo("Upgrade project is not ready yet")
+
+
+@clean_after
+def upgrade_item(ctx: Context, item_type: str, item_public_id: PublicId) -> None:
+    """
+    Upgrade an item.
+
+    :param ctx: Context object.
+    :param item_type: the item type.
+    :param item_public_id: the item public id.
+    :return: None
+    """
+    agent_name = cast(str, ctx.agent_config.agent_name)
+    if not _check_upgrade_is_required(ctx, item_type, item_public_id):
+        raise click.ClickException(
+            f"Item {item_type} {item_public_id.author}/{item_public_id.name} can not be updated. Aborting..."
+        )
+
+    click.echo(
+        "Upgrading {} to '{}' for the agent '{}'...".format(
```

Comment:
```suggestion
        "Upgrading {} {} from version '{}' to '{}' for the agent '{}'...".format(
```
First is type, second is public id without version (`author/name`) then versions old and new

Reply:
done